src/main/java/edu/cornell/kfs/coa/document/CUAccountMaintainableImpl.java

```diff
@@ -153,5 +184,65 @@ public List getSections(org.kuali.rice.kns.document.MaintenanceDocument document
         
         return super.getSections(document, oldMaintainable);
     }
+    
+    @Override
+    public void processAfterCopy(MaintenanceDocument document, Map<String, String[]> parameters) {
+    	document.getNotes().add(getNewBoNote(CUKFSConstants.AccountCreateAndUpdateNotePrefixes.ADD));
+    	super.processAfterCopy(document, parameters);
+    }
+    
+    @Override
+    public void processAfterNew(MaintenanceDocument document, Map<String, String[]> requestParameters) {
+    	document.getNotes().add(getNewBoNote(CUKFSConstants.AccountCreateAndUpdateNotePrefixes.ADD));
+    	super.processAfterNew(document, requestParameters);
+    }
+    
+    @Override
+	public void processAfterPost(MaintenanceDocument document, Map<String, String[]> requestParameters) {
+		super.processAfterPost(document, requestParameters);
+		String statusCode = document.getDocumentHeader().getWorkflowDocument().getStatus().getCode();
+		boolean isPreRoute = DocumentStatus.INITIATED.getCode().equals(statusCode)
+				|| DocumentStatus.SAVED.getCode().equals(statusCode);
+
+		if (isPreRoute) {
+			// Search for edit note and update it to include the doc description
+			for (Note note : document.getNotes()) {
+				if (note.getNoteText().startsWith(CUKFSConstants.AccountCreateAndUpdateNotePrefixes.EDIT + " account document ID " + getDocumentNumber())) {
+					note.setNoteText(CUKFSConstants.AccountCreateAndUpdateNotePrefixes.EDIT + " account document ID " + getDocumentNumber() + " " + document.getDocumentHeader().getDocumentDescription());
+				}
+			}
+		}
+	}
+    
+    /**
+     * creates a new bo note and sets the timestamp.
+     *
+     * @return a newly created note
+     */
+    protected Note getNewBoNote(String prefix){
+        Note newBoNote = new Note();
+        newBoNote.setNoteText(prefix + " account document ID " + getDocumentNumber());
```

Comment:
Since this " account document ID " String is used in multiple places and is also used for note comparisons, I would suggest moving it to its own constant.